openedx/core/djangoapps/user_api/accounts/settings_views.py

```diff
@@ -59,7 +59,18 @@ def account_settings(request):
 
     """
     if should_redirect_to_account_microfrontend():
-        return redirect(settings.ACCOUNT_MICROFRONTEND_URL)
+        url = settings.ACCOUNT_MICROFRONTEND_URL
+
+        duplicate_provider = pipeline.get_duplicate_provider(messages.get_messages(request))
+        if duplicate_provider:
+            url = '{url}?{params}'.format(
+                url=url,
+                params=six.moves.urllib.parse.urlencode({
+                    'duplicate_provider': duplicate_provider,
+                }),
+            )
```

Comment:
How will this work once we no longer have the old Account Settings page/URL and have deleted this (now deprecated) code?

Reply:
I'm not sure. The redirect to the accounts settings MFE will probably need to move upstream into the TPA pipeline code.

Reply:
Interesting.  Do you know where that code is and shall we move this to that layer at this time?

Reply:
I don't think that's something I want to tackle at this time.

Reply:
If we merge as is, we can move forward to unblock the Account Settings work.   But leaving it this way will come back to bite us in the future - so it would be good to assess where to move this logic.

Reply:
I can create a Jira ticket so we don't forget about it when it comes time to remove the old views.

Reply:
Ok.

Reply:
I created https://openedx.atlassian.net/browse/ARCH-811.